### PR TITLE
Explorer: [2670] Application page indexer data

### LINKED ISSUE #2670: Explorer application page: fetch application details from the Indexer V2
starting point: https://github.com/gitcoinco/grants-stack/pull/2712 

The goal of this ticket is to finish PR #2712 and have all the data used in the application page to be loaded from the Indexer V2 removing the subgraph dependency. 
All the Metadata is already present in the indexer so we no longer need to fetch files directly from ipfs.

We keep the queries outside of the data-layer and we move them to the data-layer in the next ticket.

Sample application page: https://explorer.gitcoin.co/#/round/424/0xd4cc0dd193c7dc1d665ae244ce12d7fab337a008/0xd4cc0dd193c7dc1d665ae244ce12d7fab337a008-55

Based on `config.allo.version` we will fetch applications for allo v1 or v2:

* [ ] if the version is `allo-v1` we only fetch applications tagged with `allo-v1`
* [ ] if the version is `allo-v2` we fetch both `allo-v1` and `allo-v2` applications
* [ ] if the version is `allo-v2` we can show `allo-v1` but we cannot add the project to the cart cause we don't support v1 and v2 contracts interactions together

### PR DESCRIPTION
**Pull request by carlbarrdahl**
- Dependent on  #2671
- Fixes #2670 

What to test: project page in explorer. I tested on my local.

- [x] Includes only changes relevant to the original ticket. Significant refactoring needs to be separated.
- [x] Doesn't disable eslint rules.
- [x] Doesn't work around the type checker (including but not limited to: type casts, non-null assertions, `@ts-ignore`, unjustified optional values).
- [x] Doesn't contain commented out code.
- [x] If adding/updating a feature, it adds/updates its test script on Notion.

### CODE REVIEW

File: packages/grant-explorer/src/features/round/__tests__/ViewProjectDetails.test.tsx [hunk 1]
```diff
@@ -294,19 +316,19 @@ describe("<ViewProjectDetails/>", () => {
...
-
   it("shows an add-to-cart button", async () => {
     renderWithContext(<ViewProjectDetails />, {
       roundState: {
-        rounds: [roundWithProjects],
+        rounds: [],
         isLoading: false,
       },
     });
+    screen.logTestingPlaygroundURL();
```
Comment (boudra) [inline packages/grant-explorer/src/features/round/__tests__/ViewProjectDetails.test.tsx:331]:
```suggestion
```

File: packages/grant-explorer/src/features/projects/hooks/useApplication.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
+import useSWR from "swr";
```
Comment (boudra) [inline packages/grant-explorer/src/features/projects/hooks/useApplication.ts:1]: i am assuming this is a first task and the next one we'll me moving this to datalayer

**Review by boudra — APPROVED** [review]

File: packages/grant-explorer/src/features/projects/hooks/useApplication.ts (hunk 2 above)
Comment (boudra) [reply packages/grant-explorer/src/features/projects/hooks/useApplication.ts:1]: didn't mean the hooks, but the graphql queries and co

Comment (vacekj) [reply packages/grant-explorer/src/features/projects/hooks/useApplication.ts:1]: That's right.

**Review by gravityblast — APPROVED** [review]

**Pull request merged**